livekit-plugins/livekit-plugins-anthropic/livekit/plugins/anthropic/llm.py

```diff
@@ -0,0 +1,471 @@
+
+        # current function call that we're waiting for full completion (args are streamed)
+        self._tool_call_id: str | None = None
+        self._fnc_name: str | None = None
+        self._fnc_raw_arguments: str | None = None
+
+    async def aclose(self) -> None:
+        if self._anthropic_stream:
+            await self._anthropic_stream.close()
+
+        return await super().aclose()
+
+    async def __anext__(self):
+        if not self._anthropic_stream:
+            self._anthropic_stream = await self._awaitable_anthropic_stream
+
+        async for event in self._anthropic_stream:
+            if event.type == "message_start":
+                pass
+            elif event.type == "message_delta":
+                pass
+            elif event.type == "message_stop":
+                pass
+            elif event.type == "content_block_start":
+                if event.content_block.type == "tool_use":
+                    self._tool_call_id = event.content_block.id
+                    self._fnc_raw_arguments = ""
+                    self._fnc_name = event.content_block.name
+            elif event.type == "content_block_delta":
+                delta = event.delta
+                if delta.type == "text_delta":
+                    return llm.ChatChunk(
+                        choices=[
+                            llm.Choice(
+                                delta=llm.ChoiceDelta(
+                                    content=delta.text, role="assistant"
+                                )
+                            )
+                        ]
+                    )
+                elif delta.type == "input_json_delta":
+                    assert self._fnc_raw_arguments is not None
+                    self._fnc_raw_arguments += delta.partial_json
+            elif event.type == "content_block_stop":
+                if self._tool_call_id is not None and self._fnc_ctx:
+                    assert self._fnc_name is not None
+                    assert self._fnc_raw_arguments is not None
+                    fnc_info = _create_ai_function_info(
+                        self._fnc_ctx,
+                        self._tool_call_id,
+                        self._fnc_name,
+                        self._fnc_raw_arguments,
+                    )
+                    self._function_calls_info.append(fnc_info)
+                    chunk = llm.ChatChunk(
+                        choices=[
+                            llm.Choice(
+                                delta=llm.ChoiceDelta(
+                                    role="assistant", tool_calls=[fnc_info]
+                                ),
+                                index=0,
+                            )
+                        ]
+                    )
+                    self._tool_call_id = None
+                    self._fnc_raw_arguments = None
+                    self._fnc_name = None
+                    return chunk
+
+        raise StopAsyncIteration
```

Comment:
I think this isn't needed

Reply:
I think it is no? Because it is in `__anext__`